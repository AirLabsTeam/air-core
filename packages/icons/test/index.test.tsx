import React from 'react';
import { render } from '@testing-library/react';
// import { theme } from '../../zephyr/src';
import Icon from '../dist';

describe('Icons', () => {
  it('uses Edit icon', () => {
    const component = render(<Icon name="Edit" />);
    const tree = component.container.firstChild;
    expect(tree).toMatchSnapshot();
  });

  it('uses Edit icon with circular border', () => {
    const component = render(<Icon name="Edit" border />);
    const tree = component.container.firstChild;
    expect(tree).toMatchSnapshot();
  });

  it('uses Edit icon with a red background', () => {
    const component = render(<Icon name="Edit" backgroundColor="red" />);
    const tree = component.container.firstChild;
    expect(tree).toMatchSnapshot();
  });

  it('uses Edit icon with a stroke width of one', () => {
    const component = render(<Icon name="Edit" strokeWidth={1} />);
    const tree = component.container.firstChild;
    expect(tree).toMatchSnapshot();
  });
});
