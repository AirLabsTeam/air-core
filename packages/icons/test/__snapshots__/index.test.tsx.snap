// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons uses Edit icon 1`] = `
<div
  style="box-sizing: border-box; display: inline-block; vertical-align: middle; line-height: 0; width: 32px; height: 32px; border-width: 0px; border-style: solid; border-color: transparent; border-radius: 0; padding: 0px; background-color: transparent;"
>
  <svg
    aria-label="[title]"
    height="32"
    role="img"
    viewBox="0 0 32 32"
    width="32"
  >
    <title>
      Edit
    </title>
    <path
      d="M13,26,4,28l2.2-8.6,14-14a4.6,4.6,0,0,1,6.5,0h0a4.6,4.6,0,0,1,0,6.5ZM18,7.5,24.5,14"
      fill="none"
      stroke="#000000"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons uses Edit icon and is filled black 1`] = `
<div
  style="box-sizing: border-box; display: inline-block; vertical-align: middle; line-height: 0; width: 32px; height: 32px; border-width: 0px; border-style: solid; border-color: transparent; border-radius: 0; padding: 0px; background-color: transparent;"
>
  <svg
    aria-label="[title]"
    height="32"
    role="img"
    viewBox="0 0 32 32"
    width="32"
  >
    <title>
      Edit
    </title>
    <path
      d="M13,26,4,28l2.2-8.6,14-14a4.6,4.6,0,0,1,6.5,0h0a4.6,4.6,0,0,1,0,6.5ZM18,7.5,24.5,14"
      fill="#000000"
      stroke="#000000"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons uses Edit icon with a red background 1`] = `
<div
  style="box-sizing: border-box; display: inline-block; vertical-align: middle; line-height: 0; width: 32px; height: 32px; border-width: 0px; border-style: solid; border-color: transparent; border-radius: 0; padding: 0px; background-color: red;"
>
  <svg
    aria-label="[title]"
    height="32"
    role="img"
    viewBox="0 0 32 32"
    width="32"
  >
    <title>
      Edit
    </title>
    <path
      d="M13,26,4,28l2.2-8.6,14-14a4.6,4.6,0,0,1,6.5,0h0a4.6,4.6,0,0,1,0,6.5ZM18,7.5,24.5,14"
      fill="none"
      stroke="#000000"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</div>
`;

exports[`Icons uses Edit icon with a stroke width of one 1`] = `
<div
  style="box-sizing: border-box; display: inline-block; vertical-align: middle; line-height: 0; width: 32px; height: 32px; border-width: 0px; border-style: solid; border-color: transparent; border-radius: 0; padding: 0px; background-color: transparent;"
>
  <svg
    aria-label="[title]"
    height="32"
    role="img"
    viewBox="0 0 32 32"
    width="32"
  >
    <title>
      Edit
    </title>
    <path
      d="M13,26,4,28l2.2-8.6,14-14a4.6,4.6,0,0,1,6.5,0h0a4.6,4.6,0,0,1,0,6.5ZM18,7.5,24.5,14"
      fill="none"
      stroke="#000000"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1"
    />
  </svg>
</div>
`;

exports[`Icons uses Edit icon with circular border 1`] = `
<div
  style="box-sizing: border-box; display: inline-block; vertical-align: middle; line-height: 0; width: 32px; height: 32px; border-width: 2px; border-style: solid; border-color: #000000; border-radius: 512px; padding: 4px; background-color: transparent;"
>
  <svg
    aria-label="[title]"
    height="20"
    role="img"
    viewBox="0 0 32 32"
    width="20"
  >
    <title>
      Edit
    </title>
    <path
      d="M13,26,4,28l2.2-8.6,14-14a4.6,4.6,0,0,1,6.5,0h0a4.6,4.6,0,0,1,0,6.5ZM18,7.5,24.5,14"
      fill="none"
      stroke="#000000"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="3"
    />
  </svg>
</div>
`;
